src/EditorFeatures/CSharp/UseAutoProperty/UseAutoPropertyAnalyzer.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.UseAutoProperty;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UseAutoProperty
+{
+    [Export]
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class UseAutoPropertyAnalyzer : AbstractUseAutoPropertyAnalyzer<PropertyDeclarationSyntax, FieldDeclarationSyntax, VariableDeclaratorSyntax, ExpressionSyntax>
+    {
+        protected override bool IsLanguageVersionSupported(Compilation compilation)
+        {
+            return ((CSharpCompilation)compilation).LanguageVersion >= LanguageVersion.CSharp6;
+        }
+
+        protected override void RegisterIneligibleFieldsAction(CompilationStartAnalysisContext context, ConcurrentBag<IFieldSymbol> ineligibleFields)
+        {
+            context.RegisterSyntaxNodeAction(snac => AnalyzeArgument(ineligibleFields, snac), SyntaxKind.Argument);
+        }
+
+        private void AnalyzeArgument(ConcurrentBag<IFieldSymbol> ineligibleFields, SyntaxNodeAnalysisContext context)
+        {
+            // An argument will disqualify a field if that field is used in a ref/out position.  
+            // We can't change such field references to be property references in C#.
+            var argument = (ArgumentSyntax)context.Node;
+            if (argument.RefOrOutKeyword.Kind() == SyntaxKind.None)
+            {
+                return;
+            }
+
+            var cancellationToken = context.CancellationToken;
+            var symbolInfo = context.SemanticModel.GetSymbolInfo(argument.Expression, cancellationToken);
+            AddIneligibleField(symbolInfo.Symbol, ineligibleFields);
+            foreach (var symbol in symbolInfo.CandidateSymbols)
```

Comment:
Hrmm.. we'll only have candidates in this case when your code is busted.  Likely meaning that there is some conflict with that field (i.e. another member with teh same name).  In that case, i'm wary about offering this fix for that field since i don't want o screw up your code any more.